### ISSUE TITLE
Still javadoc errors with JDK 11

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#304]
javadoc since JDK 9-11 is still a buggy mess:
https://bugs.openjdk.java.net/browse/JDK-8208269
We need to upgrade to `11.0.5`

**Comment by hohwille** [comment]
Do they actually test javadoc before creating releases?
https://bugs.openjdk.java.net/browse/JDK-8212233

**Comment by hohwille** [comment]
https://issues.apache.org/jira/browse/MJAVADOC-622

**Comment by markusschuh** [comment]
I have not tested recently with java 11 `<11.0.5`. But I can add, it works with `11.0.5`

The details:

The step for "devonfw-ide-configurator" first failed when I just had run
`mvn javadoc:javadoc`
which yields:

```
[ERROR] Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in https://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.
[ERROR]
[ERROR] Command line was: "C:\Program Files\AdoptOpenJDK\jdk-11.0.5.10-hotspot\bin\javadoc.exe" @options @packages
```

After debugging the javadoc call, I recognized that an additional `--release 8` or `-source 8` option for the javadoc call is necessary due to the link to javax.json. After that I realized,  there is a already a configuration for this additional compatibility flag within the "deploy" profile inside the pom.xml of `devonfw-ide`.

So I could make maven to successfully complete this javadoc generation by running
`mvn clean javadoc:javadoc -P deploy`

Since this runs into other issues, I assume, I do not call mvn yet as it is intended here. But at least, the javadoc generation issue should be handable.

N.B.: Worth to mention, I do not have used the tool versions from `devonfw-ide` itself here, but instead:
```
Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T17:06:16+02:00)
Maven home: C:\ProgramData\chocolatey\lib\maven\apache-maven-3.6.2\bin\..
Java version: 11.0.5, vendor: AdoptOpenJDK, runtime: C:\Program Files\AdoptOpenJDK\jdk-11.0.5.10-hotspot
```
